hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -243,7 +244,14 @@ private void resetFileGroupsReplaced(HoodieTimeline timeline) {
       }
     });
 
-    Map<HoodieFileGroupId, HoodieInstant> replacedFileGroups = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    // Duplicate key error when insert_overwrite same partition in multi writer, distinct HoodieFileGroupId,If repeated, save instant with the larger timestamp
+    Map<HoodieFileGroupId, HoodieInstant> replacedFileGroups = new HashMap<>();
+    resultStream.forEach(entry -> {
+      boolean containsKey = replacedFileGroups.containsKey(entry.getKey());
```

Comment:
Use a merge function in `Collectors.toMap` to select the original key-value.

Reply:
ok，got it

Reply:
OK，I'm going to change this code

Reply:
I have updated this code, please help to review it. thanks a lot